File: java/soma10numeros/soma10numeros.java
import java.util.stream.IntStream;
import java.util.stream.*; 
import java.util.List;

class Soma10Numeros {
    public static void main(String[] args){
        List<Integer> lista = IntStream.iterate(1, i -> i < 11, i -> i + 1)
            .boxed()
            .collect(Collectors.toList());
        
        var total = 0;

        for(Integer item : lista){
           total += item;
        }
        
        System.out.println("Na mão " + total);

        Integer automatico =  lista
            .stream()
            .reduce(0, (subtotal, element) -> subtotal + element);

        System.out.println("Automático " + automatico);

    }
}

File: ocaml/soma10numeros/soma10numeros.ml
let rec range a b =
    if a > b then []
    else a :: range (a+1) b;;

let lista =
    range 1 10;;

let rec soma l =
match l with
    [] -> 0
    |h::t -> h+ (soma t);;

let total  = 
    soma lista

let () =
    Printf.printf "Na mão %d \n" total;;
